app/components/govuk_component/pagination_component.rb

```diff
@@ -0,0 +1,191 @@
+    PreviousItem.new(href: href, text: text, label: label, block_mode: block_mode?, classes: classes, html_attributes: html_attributes)
+  end
+
+  renders_one :next_item, ->(href:, text: "Next", label: nil, classes: [], html_attributes: {}) do
+    NextItem.new(href: href, text: text, label: label, block_mode: block_mode?, classes: classes, html_attributes: html_attributes)
+  end
+
+  renders_many :items, "PageItem"
+
+  def initialize(pagy: nil, classes: [], html_attributes: {})
+    @pagy = pagy
+
+    super(classes: classes, html_attributes: html_attributes)
+  end
+
+  def call
+    tag.nav(class: classes, role: "navigation", aria: { label: "results" }, **html_attributes) do
+      tag.ul(class: "govuk-pagination__list") do
+        safe_join([
+          previous_content,
+          page_items,
+          next_content
+        ].flatten)
+      end
+    end
+  end
+
+private
+
+  def default_classes
+    class_names(
+      "govuk-pagination",
+      "govuk-pagination--block" => block_mode?
+    ).split
+  end
+
+  def block_mode?
+    pagy.nil? && items.none?
+  end
+
+  def previous_content
+    @previous_content ||= if pagy.present? && pagy.prev
+                            render(PreviousItem.new(href: pagy_url_for(pagy, pagy.prev)))
+                          else
+                            previous_item
+                          end
+  end
+
+  def next_content
+    @next_content ||= if pagy.present? && pagy.next
+                        render(NextItem.new(href: pagy_url_for(pagy, pagy.next), block_mode: block_mode?))
+                      else
+                        next_item
+                      end
+  end
+
+  def page_items
+    items.presence || build_items
+  end
+
+  def build_items
+    return unless pagy
+
+    pagy.series.map { |i| render(PageItem.new(number: i, href: pagy_url_for(pagy, i))) }
+  end
+
+  class AdjacentItem < GovukComponent::Base
+    include Pagy::UrlHelpers
+    attr_reader :text, :href, :label, :suffix, :block_mode
+    alias_method :block_mode?, :block_mode
+
+    def initialize(href:, suffix:, label: nil, text: nil, block_mode: false, classes: [], html_attributes: {})
+      @href       = href
+      @label      = label
+      @text       = text
+      @suffix     = suffix
+      @block_mode = block_mode
+
+      super(classes: classes, html_attributes: html_attributes)
+    end
+
+    def call
+      tag.li(class: ["govuk-pagination__item", "govuk-pagination__item--#{suffix}"]) do
+        tag.a(href: href, class: %w(govuk-link govuk-pagination__link), rel: suffix) do
+          safe_join([
+            tag.span(class: title_classes) { safe_join(body) },
+            accessible_label
+          ])
+        end
+      end
+    end
+
+  private
+
+    def title_classes
+      class_names(
+        "govuk-pagination__link-title",
+        # "govuk-pagination__link-title--decorated" => block_mode?
```

Comment:
Not really sure what decoration is here or when it should be applied.

https://github.com/alphagov/govuk-frontend/pull/2610/files#diff-8f24600b8b58d8a6ff19d90cadc27de09b8314b4405af61913e4e509858d39a4R107